.github/workflows/release.yaml

```diff
@@ -0,0 +1,46 @@
+name: Release
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  build:
+    name: Build and release SemTK
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - shell: bash
+        name: Build SemTK
+        run: |
+          . .env
+          # TODO: The tests currently fail in Github Actions.
+          mvn --quiet --batch-mode clean install -DskipTests
+
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: true
+          prerelease: false
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF#refs/tags/}
+
+      - name: Upload release assets
+        id: upload_release_asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: "distribution/target/semtk-opensource-${{ steps.get_version.outputs.VERSION }}-bin.tar.gz"
+          asset_name: semtk-opensource-${{ steps.get_version.outputs.VERSION }}-bin.tar.gz
+          asset_content_type: application/gzip
```

Comment:
Question for Langston - I looked at your repo fork's releases page, but none of your recent releases have the distribution tarball attached to them.  Did you remove the tarball manually after testing that these steps worked?

Question for Paul - do we want to create a release on the Releases page and attach the tarball there?  Or do we want to deploy to Sonatype so that snapshots will be available there and releases will be sync'ed to Maven central too?

Reply:
The very latest one has a distribution tarball attached:

https://github.com/langston-barrett/semtk/releases/tag/untagged-826ce881b5960e036f88

It's only the latest one because that's the only tag which generates an `asset_path` in this step that actually matches a build product and successfully uploads. Does that make sense?

Reply:
Oh, your release is still in draft mode and hasn't been published yet.  I get a 404 when I click on your release link and I don't see your latest release in <https://github.com/langston-barrett/semtk/releases/>, only 2.2.2-SNAPSHOT (7cfc1b0).

Reply:
Ah, I didn't realize that draft releases weren't publicly visible! This link should work: https://github.com/langston-barrett/semtk/releases/tag/2.2.2-SNAPSHOT

> Question for Paul - do we want to create a release on the Releases page and attach the tarball there? Or do we want to deploy to Sonatype so that snapshots will be available there and releases will be sync'ed to Maven central too?

My two cents: Why not both? They both have advantages for users.

- Github releases are easily accessible from the repo via the web interface, and are easy to set up with Actions. For users who are also using Github for other purposes in their CI, it's one fewer external service that has to be up for their pipelines/builds to work.
- Sonatype/Maven will be more familiar to a different audience (Java developers). For consumers of the SemTK Java APIs, this could result in one fewer external service that has to be up for their builds/pipelines to work.

The big disadvantage seems to be maintenance: Taking the time to _publish_ the Github releases, and repairing the upload steps if they ever become stale.